Improve documentation (#80)
nautobot/cookiecutter-nautobot-app: nautobot-app/{{ cookiecutter.project_slug }}/docs/dev/contributing.md

## Branching Policy

The branching policy includes the following tenets:

> - The develop branch is the branch of the next major and minor paired version planned.
> - PRs intended to add new features should be sourced from the develop branch.
> 
> {{ cookiecutter.verbose_name }} will observe semantic versioning, as of 1.0. This may result in an quick turn around in minor versions to keep pace with an ever growing feature set.

Reviewer: ```suggestion
- The `develop` branch is the branch of the next major and minor paired version planned.
- PRs intended to add new features should be sourced from the `develop` branch.
- PRs intended to fix issues in the Nautobot LTM compatible release should be sourced from the latest `ltm-<major.minor>` branch instead of `develop`.

{{ cookiecutter.verbose_name }} will observe semantic versioning, as of 1.0. This may result in a quick turnaround in minor versions to keep pace with an ever growing feature set.
```

## Release Policy

{{ cookiecutter.verbose_name }} has currently no intended scheduled release schedule, and will release new features in minor versions.

When a new release, from `develop` to `main`, is created the following should happen.

> - A release PR is created with:

Reviewer: ```suggestion
- A release PR is created from `develop` with:
```

>   - Update to the changelog in `docs/admin/release_notes/version_<major>.<minor>.md` file to reflect the changes.

Reviewer: ```suggestion
  - Update the release notes in `docs/admin/release_notes/version_<major>.<minor>.md` file to reflect the changes.
```

>   - Change the version from `<major>.<minor>.<patch>-beta` to `<major>.<minor>.<patch>` in pyproject.toml.

Reviewer: ```suggestion
  - Change the version from `<major>.<minor>.<patch>-beta` to `<major>.<minor>.<patch>` in `pyproject.toml`.
```

>   - Set the PR to the proper branch `main`.

Reviewer: ```suggestion
  - Set the PR to the `main` branch.
```

- Ensure the tests for the PR pass.
- Merge the PR.
- Create a new tag:
  - The tag should be in the form of `v<major>.<minor>.<patch>`.
  - The title should be in the form of `v<major>.<minor>.<patch>`.
  - The description should be the changes that were added to the `version_<major>.<minor>.md` document.
- If merged into `main`, then push from `main` to `develop`, in order to retain the merge commit created when the PR was merged

> - A post release PR is created with.

Reviewer: Are we actually doing this part today? @whitej6

Reviewer 2: Not consistently but should be
